aten/src/TH/generic/THTensorEvenMoreMath.cpp

```diff
@@ -67,10 +67,11 @@ void THTensor_(nonzero)(THLongTensor *subscript, THTensor *tensor)
   delete [] idx;
 }
 
-#if !defined(TH_REAL_IS_BOOL) /* non bool only part */
-
 void THTensor_(maskedFill)(THTensor *tensor, THByteTensor *mask, scalar_t value)
 {
+  AT_WARN("Please, note that this behavior is about to change. " \
```

Comment:
to the above point about replacements, you want the deprecation warning to say what the user should do instead.  So, something like enable bool masks and then say something like:
"masked_fill_ received a mask with dtype torch.uint8, this behavior is now deprecated, please use dtype torch.bool for masks instead."